src/exception/exception-api.ts

```diff
@@ -115,15 +104,14 @@ class ExceptionController {
       records: entityExceptionRecords,
     });
 
-    const status = !result.success ? 422 : 201;
-    return res.status(status).send(result);
+    return res.status(200).send(result);
   }
 
   @HasFullReadAccess()
   async getEntityException(req: Request, res: Response) {
     const programId = req.params.programId;
     const result = await exceptionService.operations.getEntityException({ programId });
-    return res.status(getResStatus(result)).send(result);
+    return res.status(result.success ? 200 : 404).send(result);
```

Comment:
In all other responses, a `400` was the result of not having a `result.success`.
Is there a non-obvious purpose for this one being different?
Assuming there's none, I'd change this, both for consistency and semantics.

Reply:
You're right. Personally, I agree that 400 is not the right approach here; as though it could work as a "give me something to work with, it does still sound more like an error than an empty response.

However, and based on the HTTP spec, I've always been under the impression 404 is descriptive of the server itself (https://www.rfc-editor.org/rfc/rfc9110.html#name-404-not-found); and it means "(server) resource not available", as in the endpoint itself is hidden or hasn't been implemented. A server "problem", not an "I could do my work with the endpoint you asked, but found no data". 404 gives the impression of a broken url (because that's what's semantically intended).

For that "no data", I believe the HTTP spec defined 204 "No content" (https://www.rfc-editor.org/rfc/rfc9110.html#name-204-no-content).
from the spec: 
```
indicates that the server has successfully fulfilled the request and that there is no additional content to send in the response content
```
Not finding an exception could better be 204, as in 200 without anything to respond with.

Reply:
Found this as an interesting reference on the topic: https://apihandyman.io/empty-lists-http-status-code-200-vs-204-vs-404/. Being a big fan of semantics (to a pedantic fault), I would love to discuss it further.